src/types.jl

```diff
@@ -169,7 +194,10 @@ end
 TermInterface.children(x::BasicSymbolic) = arguments(x)
 TermInterface.sorted_children(x::BasicSymbolic) = sorted_arguments(x)
 function TermInterface.arguments(x::BasicSymbolic)
-    @compactified x::BasicSymbolic begin
+    arguments(x.expr)
```

Comment:
This should take `arguments(x.expr)` and wrap them in a `BasicSymbolic` with the corresponding metadata from `x.meta`, otherwise this breaks existing code. `metadata_children` would also not be necessary.

Reply:
OK. Thank you